File: src/main/java/com/koddy/server/coffeechat/domain/repository/query/response/CoffeeChatCountPerCategory.java
package com.koddy.server.coffeechat.domain.repository.query.response;

import com.querydsl.core.annotations.QueryProjection;

public record CoffeeChatCountPerCategory(
        long waiting,
        long suggested,
        long scheduled,
        long passed
) {
    @QueryProjection
    public CoffeeChatCountPerCategory {
    }
}

File: src/main/java/com/koddy/server/coffeechat/domain/repository/query/response/CoffeeChatCountPerCategory.kt
package com.koddy.server.coffeechat.domain.repository.query.response

data class CoffeeChatCountPerCategory(
    val waiting: Long = 0L,
    val suggested: Long = 0L,
    val scheduled: Long = 0L,
    val passed: Long = 0L,
) {
    companion object {
        fun zero(): CoffeeChatCountPerCategory = CoffeeChatCountPerCategory()
    }
}
